java/src/GetHistogram.java

```diff
@@ -0,0 +1,119 @@
+import niftijio.*;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringTokenizer;
+import java.util.Arrays;
+import java.util.Hashtable;
+import java.io.IOException;
+import java.io.PrintWriter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.Reducer;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+import org.apache.hadoop.util.GenericOptionsParser;
+import org.apache.hadoop.util.StringUtils;
+
+public class GetHistogram {
+
+    public static class VoxelValueMapper extends Mapper<Object, Text, IntWritable, IntWritable> {
+        
+
+        public void map(Object key, Text values, Context context) throws IOException, InterruptedException {
+            try{
+                
+                Hashtable<Integer, Integer> histogram = new Hashtable<Integer, Integer>();
+
+                //read nii.tar.gz image located at filepath
+                NiftiVolume volume = NiftiVolume.read(values.toString());
+
+                int nx = volume.header.dim[1];
+                int ny = volume.header.dim[2];
+                int nz = volume.header.dim[3];
+                int dim = volume.header.dim[4];
+                
+
+                //iterate through all voxels in image and check if voxel value already exists. If so, increment count of voxel value in hashtable by 1. Otherwise, add voxel value to hash
+                //table with initial count of 1 
+                for(int d = 0; d < dim; d++){
+                    for(int k = 0; k < nz; k++){
+                        for (int j = 0; j < ny; j++){
+                            for (int i = 0; i < nx; i++){
+                                if(histogram.containsKey((int) volume.data.get(i, j, k, d))){
```

Comment:
Good point. I am going to mention it in the readme for now, but I think will also implement it later.

As for the data types, I switched it to double - which would accept more, but still not all, and added a catch for the data types that cannot be cast . Not sure how much I like my current solution though...